sw/SoC/README.md

```diff
@@ -1 +1,84 @@
-TBD
\ No newline at end of file
+# UninaSoC Software Compilation and Usage
+
+This repository contains the software infrastructure needed to build bare-metal applications for UninaSoC.
+**Note**: For Linux-based configurations, please refer to the appropriate documentation, as this tree does not apply.
+
+All example applications, as well as custom projects, are built upon the `projects/template` project.
+Projects rely on a common set of files in the `common` directory.
+
+* The `startup.s` that implements the very basic initialization operations
+* The `UninaSoC.ld`, automatically generated during the configuration flow (see the root [README](../../README.md)).
+* The `Makefile`, that implements all basic targets for building, shared among bare-metal applications.
+
+It is expected that libraries and projects depend at least on the common files, along with a `main.c` file, which users can customize.
+
+## Build examples
+
+to build the `examples`, run
+```
+make build_examples
+```
+The existing examples include:
+- `blinky` - Supported only on the embedded configuration.
+- `echo` and `hello_world` - Supported on both embedded and HPC configurations.
+
+`echo` and `hello_world` examples use the [tinyio](https://github.com/Granp4sso/TinyIO-library-for-printf-and-scanf-) library to support `printf()` and `scanf()` on UART. 
+
+## Create a new project
+
+To create a new application project, make a copy of the `template` directory and rename it accordingly to your application name.
+
+The tree must have the following structure:
+```
+project_name
+├── ld
+│   └── user.ld
+├── Makefile
+├── inc
+└── src
+    └── main.c
+```
+
+To add user-defined code, place source files in the `src` directory and header files in the `inc` directory.
+
+### User-defined Makefile
+
+The `Makefile` in the project folder is a user-defined Makefile, that imports the common `Makefile`.
+We assume that the RISC-V toolchain is in your PATH. By default, it compiles 32-bit code with IMAD extensions.
```

Comment:
> By default, it compiles 32-bit code with IMAD extensions.
This is only true for tinyIO, right?

Reply:
Si nu scem

Reply:
achso

Reply:
tinyIO requires IMAD, therefore we set our default to IMAD. However, other examples might as well compile with I alone

Reply:
In our current structure, IMAD is not our default, since the `common/Makefile` does not define the `march` flag at all. 

So far, extensions are project-/example-specific. This is fine with me.

I suggest to just clear the line:
> By default, it compiles 32-bit code with IMAD extensions.

Reply:
Sounds ok to me